packages/core/src/common/audioPlayer.ts

```diff
@@ -0,0 +1,112 @@
+import { Howl } from 'howler';
+import { leastSevere, mostSevere, defaultAudioSrc } from './constants';
+import { AudioPlayerInterface, AudioPlayerConfig } from './types';
+
+export const calculateSeverity = (severity: number): number => {
+  return Math.max(Math.min(severity, leastSevere), mostSevere);
+};
+
+export const calculateMaxVolume = (maxVolume: number): number => {
+  return Math.max(Math.min(maxVolume, 1.0), 0.0);
+};
+export class AudioPlayer implements AudioPlayerInterface {
+  config: AudioPlayerConfig;
+  private soundID: number | undefined;
+  private player: Howl | undefined;
+  private severity: number | undefined;
+
+  constructor() {
+    this.config = {
```

Comment:
we don't need to do this in a constructor, we can just say `config: AudioPlayerConfig = {...}` (preferred)